src/EditorFeatures/Core/Shared/Extensions/ITextViewExtensions.cs

```diff
@@ -383,5 +384,13 @@ public static bool TryMoveCaretToAndEnsureVisible(this ITextView textView, Virtu
         /// </summary>
         public static bool IsNotSurfaceBufferOfTextView(this ITextView textView, ITextBuffer textBuffer)
             => textBuffer != textView.TextBuffer;
+
+        internal static bool IsInLspEditorContext(this ITextView textView)
+        {
+            // If any of the buffers in the projection graph are in the LSP editor context, then we consider this to be in an LSP context.
+            // We cannot be in a partial context where some buffers are LSP and some are not.
+            var anyBufferInLspContext = textView.BufferGraph.GetTextBuffers(textBuffer => textBuffer.IsInLspEditorContext()).Any();
```

Comment:
>GetTextBuffers [](http://example.com/codeflow?start=61&length=14)

GetTextBuffers does not seem to be lazy, so this will create a collection of all the buffers and then check if it is non-empty. If the predicate always returned false, but set a flag it would avoid populating the collection.